dissect/target/plugins/os/unix/log/journal.py

```diff
@@ -0,0 +1,388 @@
+    le64_t      entry_offset;
+    le64_t      entry_array_offset;
+    le64_t      n_entries;
+    char        payload[size - 64];  // Data objects carry actual field data in the payload[] array.
+};
+
+struct EntryItem {
+    uint64_t object_offset;
+    uint64_t hash;
+};
+
+// The first four members are copied from ObjectHeader, so that the size can be used as the length of items
+struct EntryObject {
+    ObjectType  type;
+    uint8_t     flags;
+    uint8_t     reserved[6];
+    le64_t      size;
+    le64_t      seqnum;
+    le64_t      realtime;
+    le64_t      monotonic;
+    sd_id128_t  boot_id;
+    le64_t      xor_hash;
+    EntryItem   items[size - 64];
+};
+
+// The first four members are copied from from ObjectHeader, so that the size can be used as the length of entry_object_offsets
+struct EntryArrayObject {
+    ObjectType  type;
+    uint8_t     flags;
+    uint8_t     reserved[6];
+    le64_t      size;
+    le64_t      next_entry_array_offset;
+    le64_t      entry_object_offsets[size - 24 / 8];
+};
+"""  # noqa: E501
+
+c_journal = cstruct()
+c_journal.load(journal_def)
+
+
+class JournalFile:
+    """Parse Systemd Journal file format.
+
+    References:
+        - https://github.com/systemd/systemd/blob/206f0f397edf1144c63a158fb30f496c3e89f256/docs/JOURNAL_FILE_FORMAT.md
+        - https://github.com/libyal/dtformats/blob/c4fc2b8102702c64b58f145971821986bf74e6c0/documentation/Systemd%20journal%20file%20format.asciidoc
+    """  # noqa: E501
+
+    def __init__(self, fh: BinaryIO):
+        self.fh = fh
+        self.header = c_journal.Header(self.fh)
+        self.signature = "".join(chr(c) for c in self.header.signature)
+        self.entry_array_offset = self.header.entry_array_offset
+
+    def entry_object_offsets(self) -> Iterator[int]:
+        """Read object entry arrays."""
+
+        offset = self.entry_array_offset
+
+        # Entry Array with next_entry_array_offset set to 0 is the last in the list
+        while offset != 0:
+            self.fh.seek(offset)
+
+            object = c_journal.ObjectHeader(self.fh)
+
+            if object.type == c_journal.ObjectType.OBJECT_ENTRY_ARRAY:
+                # After the object is checked, read again but with EntryArrayObject instead of ObjectHeader
+                self.fh.seek(offset)
+                entry_array_object = c_journal.EntryArrayObject(self.fh)
+
+                for entry_object_offset in entry_array_object.entry_object_offsets:
+                    # Check if the offset is not zero and points to nothing
+                    if entry_object_offset:
+                        yield entry_object_offset
+
+                offset = entry_array_object.next_entry_array_offset
+
+    def decode_value(self, value: bytes) -> tuple[str, str]:
+        value = value.decode().strip()
+
+        # Remove underscores _, __
+        if value[0:1] == "__":
+            value = value[2:]
+        elif value[0] == "_":
+            value = value[1:]
```

Comment:
```suggestion
       value = value.lstrip("_")
```
Would this be an option or is it possible to have say 3 `_`'s and the 3rd shouldn't be stripped?

Reply:
That is not case, the suggested function solves the current two cases.